File: TransCoder_tokenized_test_set_functions/python/CHECK_VALID_SEQUENCE_DIVISIBLE_M.py
def isPossible ( n, index, Sum, M, arr, dp ) :
    global MAX
    if index == n :
        if ( Sum % M ) == 0 :
            return True
        return False
    if dp [ index ] [ Sum ] != - 1 :
        return dp [ index ] [ Sum ]
    placeAdd = isPossible ( n, index + 1, Sum + arr [ index ], M, arr, dp )
    placeMinus = isPossible ( n, index + 1, Sum - arr [ index ], M, arr, dp )
    res = placeAdd or placeMinus
    dp [ index ] [ Sum ] = res
    return res

File: TransCoder_tokenized_test_set_functions/java/CHECK_VALID_SEQUENCE_DIVISIBLE_M.java
static boolean isPossible ( int n, int index, int sum, int M, int arr [ ], int dp [ ] [ ] ) {
  if ( index == n ) {
    if ( ( sum % M ) == 0 ) return true;
    return false;
  }
  else if ( sum < 0 || sum >= MAX ) return false;
  if ( dp [ index ] [ sum ] != - 1 ) {
    if ( dp [ index ] [ sum ] == 0 ) return false;
    return true;
  }
  boolean placeAdd = isPossible ( n, index + 1, sum + arr [ index ], M, arr, dp );
  boolean placeMinus = isPossible ( n, index + 1, sum - arr [ index ], M, arr, dp );
  boolean res = ( placeAdd || placeMinus );
  dp [ index ] [ sum ] = ( res ) ? 1 : 0;
  return res;
}